rs/ic_os/network/src/lib.rs

```diff
@@ -1,33 +1,132 @@
 use std::path::Path;
+use std::process::Command;
 
-use anyhow::Result;
+use anyhow::{Context, Result};
+use regex::Regex;
 
 use crate::systemd::generate_systemd_config_files;
+use deterministic_ips::HwAddr;
 use info::NetworkInfo;
-use ipv6::generate_ipv6_address;
-use mac_address::mac_address::{FormattedMacAddress, UnformattedMacAddress};
 
 pub mod info;
 pub mod interfaces;
-pub mod ipv6;
 pub mod systemd;
 
 /// Write SetupOS or HostOS systemd network configuration.
 /// Requires superuser permissions to run `ipmitool` and write to the systemd directory
 pub fn generate_network_config(
     network_info: &NetworkInfo,
-    generated_mac: UnformattedMacAddress,
+    generated_mac: &HwAddr,
     output_directory: &Path,
 ) -> Result<()> {
     eprintln!("Generating ipv6 address");
-    let ipv6_address = generate_ipv6_address(&network_info.ipv6_prefix, &generated_mac)?;
-    eprintln!("Using ipv6 address: {}", ipv6_address);
+    let ipv6_address = generated_mac.calculate_slaac(&network_info.ipv6_prefix)?;
+    eprintln!("Using ipv6 address: {ipv6_address}");
 
-    let formatted_mac = FormattedMacAddress::from(&generated_mac);
     generate_systemd_config_files(
         output_directory,
         network_info,
-        Some(&formatted_mac),
+        Some(generated_mac),
         &ipv6_address,
     )
 }
+
```

Comment:
The following is relocated and consolidated from the removed `mac_address` crate.